### PR TITLE
Attach caption HLS manifest to primary manifest. 

### PR DESCRIPTION
**Pull request by tpendragon**
This generates an HLS manifest for caption files (VTT) and associates them with the primary manifest.

Closes https://github.com/pulibrary/figgy/issues/6181

### CODE REVIEW

File: app/values/hls_manifest.rb [hunk 1]
```diff
@@ -7,11 +7,12 @@ class HlsManifest
...
   def self.for(file_set:, file_metadata:, as: nil, auth_token: nil)
-    return unless file_metadata.hls_manifest?
-    if as == "stream"
+    return unless file_metadata.hls_manifest? || file_metadata.caption?
+    return new(file_set: file_set, file_metadata: file_metadata, auth_token: auth_token) unless as == "stream"
+    if file_metadata.hls_manifest?
       Primary.new(file_set: file_set, file_metadata: file_metadata, auth_token: auth_token)
     else
-      new(file_set: file_set, file_metadata: file_metadata, auth_token: auth_token)
+      Caption.new(file_set: file_set, file_metadata: file_metadata, auth_token: auth_token)
```
Comment (hackartisan) [inline app/values/hls_manifest.rb:15]: This section might be more clear to me if it were like:
```
return Caption.new... if file_metadata.caption?
return unless file_metadata.hls_manifest?
if as == "stream"
  Primary...
else
  new...
end
```

File: app/nested_resources/file_metadata.rb [hunk 2]
```diff
@@ -28,6 +28,9 @@ class FileMetadata < Valkyrie::Resource
   # PDF Metadata
   attribute :page_count, Valkyrie::Types::Integer
 
+  # Caption Metadata
+  attribute :caption_language, Valkyrie::Types::String.default("eng")
```
[the remark this answers is not in the document]
Comment (tpendragon) [reply app/nested_resources/file_metadata.rb:32]: Oh, yeah, I guess so. 🤔

[the remark this answers is not in the document]
Comment (hackartisan) [reply app/nested_resources/file_metadata.rb:32]: I don't think I like this. is there a downside to having it be nil? We'd just have to set its default every time we set it in code?

File: app/values/hls_manifest.rb [hunk 3]
```diff
@@ -7,11 +7,17 @@ class HlsManifest
...
     return unless file_metadata.hls_manifest?
+    # IIIF Manifests link to a "primary" HLS manifest, which contains links to the caption
+    # manifest and the video manifest. Build it dynamically because it's easier.
     if as == "stream"
       Primary.new(file_set: file_set, file_metadata: file_metadata, auth_token: auth_token)
     else
-      new(file_set: file_set, file_metadata: file_metadata, auth_token: auth_token)
+      # If not asked for the primary manifest, just render the video HLS manifest
+      # ffmpeg created (with auth_tokens if appropriate)
+      new(file_set: file_set, file_metadata: file_metadata, auth_token: auth_token) unless as == "stream"
```
Comment (hackartisan) [inline app/values/hls_manifest.rb:20]: redundant `unless` here

**Review by hackartisan — CHANGES_REQUESTED** [review]
one small typo from the code reorganization

File: app/values/hls_manifest.rb (hunk 3 above)
Comment (tpendragon) [reply app/values/hls_manifest.rb]: Oof, thank you. Postfix conditionals are sneaky. 🥷

**Review by hackartisan — APPROVED** [review]
🎉

**Pull request merged**